app/util.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+const request = require('request-promise');
+
+// returns a Date representing tomorrow at 12pm so that
+// we can issue darksky 'time machine' requests to get an
+// idea of the weather tomorrow at a time that should be
+// relevant to someone interested in the weather
+function getTomorrowAtNoon() {
+    let tomorrow = new Date();
+    tomorrow.setDate(tomorrow.getDate() + 1);
+    tomorrow.setHours(12);
+    return tomorrow;
+}
+
+// This url can be used to grab a frame from a DDOT (DC) traffic cam!
+// This can be used to give a rough visual of the weather. The camera is
+// at the intersection of 14th St and Independence St NW. Looks like the current
+// time in milliseconds since 1970 needs to be passed in to get the latest images.
+// The RTM client doesn't allow sending attachments (see https://api.slack.com/rtm),
+// but the Slack URL unfurling functionality automatically includes the traffic cam frame!
+// See http://app.ddot.dc.gov/ for more info.
+function getTrafficCamImageUrl() {
+    const trafficlandApiUrl = 'http://api.trafficland.com/v1.5/json/video_feeds?system=ddot&key=1594c8892d7fbd18181a8a6a44958b28&region=WAS';
+    let promise = request({ uri: trafficlandApiUrl, json: true }).then((json) => {
+        if (Array.isArray(json)) {
+            let cameraObj = json.find((element) => {
+                return element.publicId === '200146'
+            });
+
+            if (cameraObj) {
+                return cameraObj;
+            } else {
+                throw 'Unable to find desired traffic camera!';
+            }
+        }
+    });
+
+    return promise;
+}
+
+const darkskyToken = process.env.DARKSKY_TOKEN || '';
+const darkskyUrlBase = `https://api.darksky.net/forecast/${darkskyToken}`;
+
+// returns a promise that resolves to the JSON representing the current weather
+// report from darksky on success
+function getCurrentWeatherReport(location) {
+    // we don't need the minutely or hourly forecast, so we'll exclude them
+    const darkskyCurrentlyUrl = `${darkskyUrlBase}/${location.latitude},${location.longitude}` + '?exclude=minutely,hourly';
+    return request({ uri: darkskyCurrentlyUrl, json: true });
+}
+
+function getTomorrowsWeatherReport(location) {
+    const darkskyTomorrowUrl = `${darkskyUrlBase}/${location.latitude},${location.longitude},` + Math.floor(getTomorrowAtNoon().getTime() / 1000 ) + '?exclude=currently,minutely,hourly';
+    return request({ uri: darkskyTomorrowUrl, json: true });
+}
+
+// groups sets of strings used to generate messages summarizing the
+// weather at some point in time
+const weatherVerbs = {
```

Comment:
Hm, good idea. I can't think of any major downsides to that approach other than the slight duplication.

It would help keep the strings generated for the messages together and easier to see what's actually generated.